Rego/AADConfig.rego

```diff
@@ -735,37 +735,40 @@ tests[{
 #
 # MS.AAD.7.4v1
 #--
-RolesWithoutLimitedExpirationPeriod[Role.DisplayName] {
-    Role := input.privileged_roles[_]
-    Rule := Role.Rules[_]
-    RuleMatch := Rule.Id == "Expiration_Admin_Assignment"
-    ExpirationNotRequired := Rule.AdditionalProperties.isExpirationRequired == false
-    MaximumDurationCorrect := Rule.AdditionalProperties.maximumDuration == "P15D"
+default PrivilegedRoleExclusions(_, _) := false
+PrivilegedRoleExclusions(PrivilegedRole, PolicyID) := true if {
+    PrivilegedRoleAssignedPrincipals := { x.PrincipalId | some x in PrivilegedRole.Assignments; x.ActivatedUsing.EndDateTime == null }
 
-    # Role policy does not require assignment expiration
-    Conditions1 := [RuleMatch == true, ExpirationNotRequired == true]
-    Case1 := count([Condition | Condition = Conditions1[_]; Condition == false]) == 0
+    AllowedPrivilegedRoleUsers := { y | some y in input.scuba_config.Aad[PolicyID].RoleExclusions.Users; y != null }
+    AllowedPrivilegedRoleGroups := { y | some y in input.scuba_config.Aad[PolicyID].RoleExclusions.Groups; y != null }
+    AllowedPrivilegedRole := AllowedPrivilegedRoleUsers | AllowedPrivilegedRoleGroups
 
-    # Role policy requires assignment expiration, but maximum duration is not 15 days
-    Conditions2 := [RuleMatch == true, ExpirationNotRequired == false, MaximumDurationCorrect == false]
-    Case2 := count([Condition | Condition = Conditions2[_]; Condition == false]) == 0
+    count(PrivilegedRoleAssignedPrincipals) > 0
+    count(PrivilegedRoleAssignedPrincipals - AllowedPrivilegedRole) != 0
+}
 
-    # Filter: only include rules that meet one of the two cases
-    Conditions := [Case1, Case2]
-    count([Condition | Condition = Conditions[_]; Condition == true]) > 0
+PrivilegedRoleExclusions(PrivilegedRole, PolicyID) := true if {
+    count({ x.PrincipalId | some x in PrivilegedRole.Assignments; x.ActivatedUsing.EndDateTime == null }) > 0
+    count({ y | some y in input.scuba_config.Aad[PolicyID].RoleExclusions.Users; y != null }) == 0
+    count({ y | some y in input.scuba_config.Aad[PolicyID].RoleExclusions.Groups; y != null }) == 0
+}
+
+PrivilegedRolesWithoutExpirationPeriod[Role.DisplayName] {
+    Role := input.privileged_roles[_]
+
+    PrivilegedRoleExclusions(Role, "MS.AAD.7.4v1") == true
 }
 
 tests[{
     "PolicyId" : "MS.AAD.7.4v1",
     "Criticality" : "Shall",
     "Commandlet" : ["Get-MgSubscribedSku", "Get-PrivilegedRole"],
-    "ActualValue" : RolesWithoutLimitedExpirationPeriod,
-    "ReportDetails" : ReportDetailsArrayLicenseWarning(RolesWithoutLimitedExpirationPeriod, DescriptionString),
+    "ActualValue" : PrivilegedRolesWithoutExpirationPeriod,
+    "ReportDetails" : ReportDetailsArrayLicenseWarning(PrivilegedRolesWithoutExpirationPeriod, DescriptionString),
```

Comment:
Function name, ReportDetailsArrayLiscenseWarning,  does not appear to be a real meaningful contribution to readability.